sdk/resourcemanagerhybrid/azure-resourcemanager/src/main/java/com/azure/resourcemanager/AzureResourceManager.java

```diff
@@ -335,48 +342,67 @@ public AzureResourceManager withDefaultSubscription() {
+            throw LOGGER.logExceptionAsError(
+                new HttpResponseException("Failed : HTTP error code : " + response.getStatusCode(), response));
         }
         String body = response.getBodyAsString().block();
         try {
-            ArmMetadata metadata = JacksonAdapter.createDefaultSerializerAdapter()
-                .deserialize(body, ArmMetadata.class, SerializerEncoding.JSON);
+            List<ArmMetadata> metadataArray = JacksonAdapter.createDefaultSerializerAdapter()
+                .deserialize(body, ARM_METADATA_ARRAY_TYPE_REFERENCE.getJavaType(), SerializerEncoding.JSON);
+
+            if (metadataArray == null || CoreUtils.isNullOrEmpty(metadataArray)) {
+                throw LOGGER.logExceptionAsError(
+                    new HttpResponseException("Failed to find metadata : " + body, response));
+            }
 
+            ArmMetadata metadata = metadataArray.iterator().next();
             AzureEnvironment azureEnvironment = new AzureEnvironment(new HashMap<String, String>() {
-                private static final long serialVersionUID = 1L;
                 {
                     put("managementEndpointUrl", metadata.authentication.audiences.get(0));
                     put("resourceManagerEndpointUrl", armEndpoint);
-                    put("galleryEndpointUrl", metadata.galleryEndpoint);
+                    put("galleryEndpointUrl", metadata.gallery);
                     put("activeDirectoryEndpointUrl", metadata.authentication.loginEndpoint);
                     put("activeDirectoryResourceId", metadata.authentication.audiences.get(0));
-                    put("activeDirectoryGraphResourceId", metadata.graphEndpoint);
-                    put("storageEndpointSuffix", armEndpoint.substring(armEndpoint.indexOf('.')));
-                    put("keyVaultDnsSuffix", ".vault" + armEndpoint.substring(armEndpoint.indexOf('.')));
+                    put("activeDirectoryGraphResourceId", metadata.graph);
+                    put("storageEndpointSuffix", "." + metadata.suffixes.storage);
+                    put("keyVaultDnsSuffix", "." + metadata.suffixes.keyVaultDns);
                 }
             });
             return azureEnvironment;
         } catch (IOException e) {
-            throw new HttpResponseException("Failed to parse metadata : " + body, response);
+            throw LOGGER.logExceptionAsError(
+                new HttpResponseException("Failed to parse metadata : " + body, response, e));
         }
     }
 
+    private static final TypeReference<List<ArmMetadata>> ARM_METADATA_ARRAY_TYPE_REFERENCE =
+        new TypeReference<List<ArmMetadata>>() {
+        };
+
     private static class ArmMetadata {
-        private String galleryEndpoint;
-        private String graphEndpoint;
+        private String name;
+        private String gallery;
+        private String graph;
+        private String graphAudience;
```

Comment:
nit: the `graphAudience` part isn't being used and isn't in the Azure Stack response.

Reply:
removed.